docs/machinepools.md

```diff
@@ -0,0 +1,106 @@
+# MachinePools
+- **Feature status:** Experimental
+- **Feature gate:** MachinePool=true
+
+> In Cluster API (CAPI) v1alpha2, users can create MachineDeployment, MachineSet or Machine custom
+> resources. When you create a MachineDeployment or MachineSet, Cluster API components react and
+> eventually Machine resources are created. Cluster API's current architecture mandates that a
+> Machine maps to a single machine (virtual or bare metal) with the provider being responsible for
+> the management of the underlying machine's infrastructure.
+
+> Nearly all infrastructure providers have a way for their users to manage a group of machines
+> (virtual or bare metal) as a single entity. Each infrastructure provider offers their own unique
+> features, but nearly all are concerned with managing availability, health, and configuration updates.
+
+> A MachinePool is similar to a MachineDeployment in that they both define
+> configuration and policy for how a set of machines are managed. They Both define a common
+> configuration, number of desired machine replicas, and policy for update. Both types also combine
+> information from Kubernetes as well as the underlying provider infrastructure to give a view of
+> the overall health of the machines in the set.
+
+> MachinePool diverges from MachineDeployment in that the MachineDeployment controller uses
+> MachineSets to achieve the aforementioned desired number of machines and to orchestrate updates
+> to the Machines in the managed set, while MachinePool delegates the responsibility of these
+> concerns to an infrastructure provider specific resource such as AWS Auto Scale Groups, GCP
+> Managed Instance Groups, and Azure Virtual Machine Scale Sets.
+
+> MachinePool is optional and doesn't replace the need for MachineSet/Machine since not every
+> infrastructure provider will have an abstraction for managing multiple machines (i.e. bare metal).
+> Users may always opt to choose MachineSet/Machine when they don't see additional value in
+> MachinePool for their use case.
```

Comment:
Could we clean up the language here and make it more directly relevant to the AWS user base? Probably don't need to discuss the history wrt to v1alpha2, and describe more directly the relation of this implementation of MachinePools to AWS ASGs.